packages/wallets/src/hooks/useAllBalanceSubscription.ts

```diff
@@ -0,0 +1,66 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useAuthorize, useBalanceSubscription } from '@deriv/api-v2';
+import Observable from '../utils/observable';
+
+type TBalance = ReturnType<typeof useBalanceSubscription>['data']['accounts'];
+
+const balanceStore = new Observable<TBalance | undefined>(undefined);
+
+/**
+ * Custom hook that manages subscription to balance changes from `balanceStore`.
+ * Retrieves initial balance and subscribes to future updates.
+ * @returns An object containing the current balance and a function to update it.
+ * @example const { data: balanceData, subscribeToAllBalance } = useAllBalanceSubscription();
+ */
+const useAllBalanceSubscription = () => {
+    const [balance, setBalance] = useState(balanceStore.get());
+    const {
+        data: balanceData,
+        isLoading: isBalanceLoading,
+        isSubscribed,
+        subscribe,
+        unsubscribe,
+    } = useBalanceSubscription();
+    const { isSuccess: isAuthorizeSuccessful } = useAuthorize();
+    useEffect(() => {
+        return balanceStore.subscribe(setBalance); // subscribe setBalance to the balance store and return the cleanup function
+    }, []);
+
+    const subscribeToAllBalance = useCallback(() => {
+        if (!isAuthorizeSuccessful) return;
+        subscribe({
+            account: 'all',
+        });
+    }, [isAuthorizeSuccessful, subscribe]);
+
+    useEffect(() => {
+        if (!isAuthorizeSuccessful || isBalanceLoading || Object.entries(balanceData).length === 0) return;
```

Comment:
It's not really necessary @wojciech-deriv. I just added  because `useBalanceSubscription` returns an empty object instead of `undefined` when the balance is loading. I think I added it before adding the check for `isBalanceLoading`.